lib/galaxy/workflow/run.py

```diff
@@ -327,7 +336,7 @@ def replacement_for_connection(self, connection, is_data=True):
                 if not replacement.is_ok:
                     raise modules.CancelWorkflowEvaluation()
             else:
-                if not replacement.collection.populated:
+                if not replacement.collection.populated:  # type: ignore
```

Comment:
I feel like an explicit cast() would be better than ``type: ignore``.

Reply:
I think there is a real issue here, `model.DatasetCollection` doesn't have a `collection` attribute.

Reply:
I pushed a commit to fix this (vetted by Marius).